huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,413 @@
+			return diag.Errorf("error deleting old tags for EVS snapshot group (%s): %s", d.Id(), err)
+		}
+		if err := updateSnapshotGroupTags(client, d.Id(), newTags.(map[string]interface{}), true); err != nil {
+			return diag.Errorf("error creating new tags for EVS snapshot group (%s): %s", d.Id(), err)
+		}
+	}
+	return resourceEvsSnapshotGroupRead(ctx, d, meta)
+}
+
+func updateSnapshotGroupTags(client *golangsdk.ServiceClient, groupID string, tags map[string]interface{}, isCreate bool) error {
+	if len(tags) == 0 {
+		return nil
+	}
+	var (
+		apiPath string
+	)
+	if isCreate {
+		apiPath = client.Endpoint + "v5/{project_id}/snapshot-groups/{snapshot_group_id}/tags/create"
+	} else {
+		apiPath = client.Endpoint + "v5/{project_id}/snapshot-groups/{snapshot_group_id}/tags/delete"
+	}
+	apiPath = strings.ReplaceAll(apiPath, "{project_id}", client.ProjectID)
+	apiPath = strings.ReplaceAll(apiPath, "{snapshot_group_id}", groupID)
+	requestBody := map[string]interface{}{
+		"tags": expandSnapshotGroupTags(tags),
+	}
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         requestBody,
+		OkCodes:          []int{200, 201, 202, 204},
+	}
+	_, err := client.Request("POST", apiPath, &requestOpt)
+	return err
+}
+
+func resourceEvsSnapshotGroupDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v5/{project_id}/snapshot-groups/{snapshot_group_id}"
+		product = "evs"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{snapshot_group_id}", d.Id())
+	requestOpt := golangsdk.RequestOpts{KeepResponseBody: true}
+	_, err = client.Request("DELETE", requestPath, &requestOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error deleting EVS snapshot group")
+	}
+
+	if err := waitingForSnapshotGroupDeleted(ctx, client, d, d.Timeout(schema.TimeoutDelete)); err != nil {
+		return diag.Errorf("error waiting for EVS snapshot group (%s) deleted: %s", d.Id(), err)
+	}
+
+	return nil
+}
+
+func waitingForSnapshotGroupDeleted(ctx context.Context, client *golangsdk.ServiceClient, d *schema.ResourceData,
+	timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED"},
+		Refresh: func() (interface{}, string, error) {
+			respBody, err := GetSnapshotGroupDetail(client, d.Id())
+			if err != nil {
+				var errDefault404 golangsdk.ErrDefault404
+				if errors.As(err, &errDefault404) {
+					return "success deleted", "COMPLETED", nil
+				}
+				return nil, "ERROR", err
+			}
+
+			status := utils.PathSearch("snapshot_group.status", respBody, "").(string)
```

Comment:
The snapshot group does not give us the status enum values, I think `error_deleting` may not be the exist status code.
Suggess removing code from line 382 to line 391.